Added overview and requirements sections for configurable products

carlisaac opened the pull request:
Added new sections and reordered existing sections.

Closes GH-1016

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/spartacus-docs?pullRequest=1017) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SAP/spartacus-docs?pullRequest=1017) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SAP/spartacus-docs?pullRequest=1017) it.</sub>

carlisaac commented:
@gladius-mtl This is now ready to merge for the 3.1 release.
- Added section on configuration
- Added overview and requirements
- Updated link in pre-release info to point to page on wiki instead of GitHub repo
